content/Basic Cryptography/DES.md

```diff
@@ -0,0 +1,191 @@
+When DES was first introduced in the 1970s, a 56-bit key length provided adequate security. However, with advancements in computing, DES is no longer considered secure. To address these limitations, multiple DES techniques were introduced, notably Triple DES (3DES), which applies DES three times and provides 112 bits of security, making it more robust.
+In 2005, NIST officially retired simple DES. Triple DES was also deprecated in 2019, and its usage was prohibited by the end of 2023, except for processing already encrypted data. Today, DES has been replaced by the more secure and powerful [[AES]].
+
+## Design
+High-level description of DES is as following:
+
+1. Initial Permutation (IP)
+2. 16 Rounds of F function and Swap
+	* F Function
+		* Expansion P-Box
+		* XOR with Round Key
+		* S-Box Substitution
+		* P-Box Permutation
+	* Swap
+1. Inverse Initial Permutation (IP⁻¹)
+2. Key Scheduling
+	* Parity Bit Drop (PC-1)
+	* Shift
+	* Compression P-Box (PC-2)
+
+### Initial Permutation (IP)
+The first step applied when 64-bit plaintext is input is the **Initial Permutation (IP)**, where the bits are rearranged based on a predefined permutation.
+
+| IP  |     |     |     |     |     |     |     |
+| :-: | :-: | :-: | :-: | :-: | :-: | :-: | :-: |
+| 58  | 50  | 42  | 34  | 26  | 18  | 10  |  2  |
+| 60  | 52  | 44  | 36  | 28  | 20  | 12  |  4  |
+| 62  | 54  | 46  | 38  | 30  | 22  | 14  |  6  |
+| 64  | 56  | 48  | 40  | 32  | 24  | 16  |  8  |
+| 57  | 49  | 41  | 33  | 25  | 17  |  9  |  1  |
+| 59  | 51  | 43  | 35  | 27  | 19  | 11  |  3  |
+| 61  | 53  | 45  | 37  | 29  | 21  | 13  |  5  |
+| 63  | 55  | 47  | 39  | 31  | 23  | 15  |  7  |
+
+The permuted input takes bit 58 of the original input as its first bit, bit 50 as its second bit, and continues in this manner, ending with bit 7 as its final bit. This permuted block then serves as the input to a key-dependent computation, which is described below.
+
+### 16 Rounds of F function and Swap
+DES consists of a total of 16 rounds, each divided into two stages: the **Mixer** stage and the **Swapper** stage.
+
+1. **Mixer Stage**: In this stage, the 32-bit right half and the round key are used to calculate the F function value, which is then XORed with the 32-bit left half. This can be represented by the following formula: $$ R_i = L_{i-1} \oplus F(R_{i-1}, k_i) $$
+2. **Swapper Stage**: In the swapper stage, the value computed from the XOR operation is placed on the right, while the original 32-bit right half is moved to the left. This can be expressed as: $$ L_i = R_{i-1} $$
+Thus, the i-th round can be represented as follows.
+$$ L_i = R_{i-1} $$
+$$ R_i = L_{i-1} \oplus F(R_{i-1}, k_i) $$
+
+#### F function
+##### Expansion P-Box
+The **Expansion P-Box** is a process that permutes the order of the bits of a 32-bit input value while expanding the output to 48 bits.
+The method for expanding the input value is as follows:
+![[Pasted image 20241103212213.png]]
```

Comment:
Change the image name to "images/DES(1)" (This is the convention in this github)